src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -49,313 +51,281 @@ public BtConnectorHelper() {
+        DisconnectIncomingConnections();
 
-            mServerSocketList.clear();
+        // disconnect outgoing connection
+        DisconnectAll ();
+    }
+
+    // we only cut off our outgoing connections, incoming ones are cut off from the other end.
+    // if we want to cut off whole communications, we'll do Stop
+    public boolean Disconnect(String peerId){
+        BtToRequestSocket tmpSoc = mBtToRequestSocket;
+        if(tmpSoc == null) {
+            return false;
         }
 
-        if(mBtToRequestSocket != null) {
-            print_debug("Disconnect:::Stop : mBtToRequestSocket");
-            mBtToRequestSocket.Stop();
+        String currenPeerId = tmpSoc.GetPeerId();
+        print_debug("Disconnect : " + peerId + ", current request : " + currenPeerId);
+        if (peerId.equalsIgnoreCase(currenPeerId)) {
+            tmpSoc.Stop();
             mBtToRequestSocket = null;
+            return true;
         }
+        return false;
     }
 
-    public boolean Disconnect(String peerId){
-
-        boolean ret = false;
-
-// we only cut off our outgoing connections, incoming ones are cut off from the other end.
-// if we want to cut off whole communications, we'll do Stop
-        if(mBtToRequestSocket != null) {
-            String currentpeerId = mBtToRequestSocket.GetPeerId();
-            print_debug("Disconnect : " + peerId + ", current request : " + currentpeerId);
-            if(peerId.length() == 0 || peerId.equalsIgnoreCase(currentpeerId)) {
-                print_debug("Disconnect:::Stop :" + currentpeerId);
-                mBtToRequestSocket.Stop();
-                mBtToRequestSocket = null;
-                ret = true;
-            }
+    private void DisconnectAll(){
+        BtToRequestSocket tmpSoc = mBtToRequestSocket;
+        mBtToRequestSocket = null;
+        if(tmpSoc != null) {
+            tmpSoc.Stop();
         }
-
-        return ret;
     }
 
-    //test time implementation to simulate 'peer disappearing'
-    public boolean  DisconnectIncomingConnections(){
+    //function to disconnect all incoming connections
+    // should only be used internally, i.e. should be private
+    // but for testing time, this is marked as public, so we can simulate 'peer disappearing'
+    // by cutting off the connection from the remote party
+    public boolean  DisconnectIncomingConnections() {
 
         boolean ret = false;
-        if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                    ret = true;
-                }
+        Iterator<BtToServerSocket> iterator = mServerSocketList.iterator();
+        while (iterator.hasNext()) {
```

Comment:
@DrJukka This isn't the idiomatic way of handling this. As IntelliJ will automatically fix for you this should be written as:

```Java
for (BtToServerSocket rSocket : mServerSocketList) {
   ...
}
```

BTW, you shouldn't need to test for null since as long as you use 'clear' you will never have a null entry. If you are testing for null it should be for defensive purposes but in that case you should throw an exception or at least log something if you find a null since that would be an error.